mkosi/__init__.py

```diff
@@ -3260,6 +3260,26 @@ def run_finalize_script(args: MkosiArgs, root: Path, do_run_build_script: bool,
         run([args.finalize_script, verb], env=env)
 
 
+def make_reproducible(args: MkosiArgs, root: Path, do_run_build_script: bool, for_cache: bool) -> None:
+    verb = "build" if do_run_build_script else "final"
+
+    with complete_step("Attempting to make build reproducible…"):
+        env = collections.ChainMap(dict(BUILDROOT=str(root), OUTPUTDIR=str(output_dir(args))),
+                                   args.environment,
+                                   os.environ)
+
+        file = f"reproducible.{args.distribution}.sh"
+        if importlib.resources.is_resource("mkosi.resources", file):
+            with importlib.resources.path("mkosi.resources", file) as p:
+                run(["bash", p, verb], env=env)
+        else:
+            warn("Reproducible images is not support on {arge.distribution}")
+
+    # TODO: Support SOURCE_DATE_EPOCH
+    reset_timestamps = ["find", root, "-mindepth", "1", "-execdir", "touch", "-hcd", "@0", "{}", "+"]
```

Comment:
```suggestion
    reset_timestamps = ["find", root, "-mindepth", "1", "-execdir", "touch", "--no-dereference", "--no-create", "--date=@0", "{}", "+"]
```
should be more readable long term.

Reply:
Can we do this without shelling out? sounds simple enough to do that within python, even if it might be a tad bit slower

Reply:
Sure, can give that a try.

Reply:
I think the whole approach of resetting the time is somewhat iffy. Why would we need to reset the time? For stuff that is installed from packages, I think we'd want to keep the installed timestamps. OTOH, it is the build tools that should honour the variable. In particular, I'm surprised that `meson` doesn't do this natively. E.g. wouldn't it be better if `meson install` touched the time on all files it installs? I would open a pull request against meson to add support for this.

I see though how a blanket reset of the time can be useful. But maybe this should be opt-in, and not actually advertised as "reproducible builds", but just as a low-level feature: `ResetFileTimes=` that'd accept a set of path globs, similarly to `RemoveFiles=` and other settings. This would be useful as a work-around for tools that don't support `SOURCE_DATE_EPOCH`.

Reply:
Patching this into `meson` doesn't solve a lot considering most projects doesn't even use it. We do need to touch the time since we are aiming for reproducible builds and the timestamps are part of the archive. Doing this in one swoop when people do request a repoducible builds is a small thing imo and doesn't need a lot more.

Having this on the side, as well as including it in the `--reproduce` switch, is fine imo. But not necessarily something that belongs in this PR.